### PR TITLE
Upgrade gp-common-go-libs to new version

### PR DESCRIPTION
**Pull request by bradfordb-vmware**
This commit was generated with the following:

```
cd ~/workspace/pxf/cli/go/src/pxf-cli
git switch -c test-gp-common-go-libs-golang-1.17
go mod edit -go=1.17
go get github.com/greenplum-db/gp-common-go-libs@golang-1.17
go mod tidy
git add -p .
git commit
```

Authored-by: Bradford D. Boyle <bradfordb@vmware.com>

### CODE REVIEW

File: cli/go/src/pxf-cli/go.mod [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
-	github.com/onsi/ginkgo v1.14.0
-	github.com/onsi/gomega v1.10.1
+	github.com/greenplum-db/gp-common-go-libs v1.0.5-0.20220119222931-572bf8fa8e9c
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.17.0
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/cobra v0.0.7
+)
+
+require (
```
Comment (denalex) [inline cli/go/src/pxf-cli/go.mod:13]: it seems this section is for indirect dependencies, so maybe put a comment for the section and remove the same comment from all individual lines ?

File: cli/go/src/pxf-cli/go.mod [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+	github.com/jackc/pgx/v4 v4.14.1 // indirect
+	github.com/jmoiron/sqlx v1.3.4 // indirect
+	github.com/nxadm/tail v1.4.8 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
+	golang.org/x/crypto v0.0.0-20220112180741-5e0467b6c7ce // indirect
+	golang.org/x/net v0.0.0-20220114011407-0dd24b26b47d // indirect
+	golang.org/x/sys v0.0.0-20220114195835-da31bd327af9 // indirect
+	golang.org/x/text v0.3.7 // indirect
+	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
```
Comment (denalex) [inline cli/go/src/pxf-cli/go.mod:34]: did we really go from 3 indirect deps to all of these ? Are they all statically linked and we have to declare them even if they are not used by our use cases ?

**Review by denalex — APPROVED** [review]

File: cli/go/src/pxf-cli/go.mod (hunk 1 above)
Comment (bradfordb-vmware) [reply cli/go/src/pxf-cli/go.mod:13]: The indirect comments are auto-generated by the `go` tool when we updated our dependencies, not sure if I can turn that off.

File: cli/go/src/pxf-cli/go.mod (hunk 2 above)
Comment (bradfordb-vmware) [reply cli/go/src/pxf-cli/go.mod:34]: I looked at this some more and it looks like most of the additional indirect dependencies are from updating to Go 1.17 (i.e., `go mod edit -go=1.17`). It looks like this is expected with Go 1.17 (see 'Pruned module graphs in go 1.17 modules' in [Go 1.17 Release Notes](https://go.dev/doc/go1.17#go-command)).

**Review by ashuka24 — APPROVED** [review]

**Pull request merged**